[Oei Yi Ping] iP

schping opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

tofu-and-noodles committed:
Duke class: Add greet() and listen() instance methods

greet() prints the designated message.

listen() accepts a user command and echoes said command.
After the printing the previous command, Duke then waits for a new
command. Active listening via a while-loop allows Duke to repeatedly
carry out the above cycle until "bye" command is received.

Upon initiating Duke object, greet() and listen() are called to
immediately greet and accept commands.

tofu-and-noodles committed:
Duke class: add list attribute; add bye(), addToList(), printList()

Add list attribute to Duke class to support String storage.

Package actions to be done upon "bye" command into bye() method,
helps with modularity and SLAP principle.

addToList() adds input String into the list attribute of Duke
object.

printList() prints all elements in list.

Change listen() implementation from if-else to switch, for better
support of handling numerous types of commands.

tofu-and-noodles committed:
Add Task class

Add Task class for better representation of added tasks, instead of
merely using String. Task class can support the various required
attributes of the tasks such as Done status.

tofu-and-noodles committed:
Duke class: add taskDone() method, change list to list of Task

Change list of String to list of Task, rename list to tasks.

Package actions to be done upon "done" command into taskDone()
method, helps with modularity and SLAP principle.

tofu-and-noodles committed:
Add subclasses of Task: ToDo, Event, Deadline

tofu-and-noodles committed:
Duke class: Add todo, event, deadline as supported commands

Add the above commands to specifically create the corresponding
object.

taskDone(): fix off-by-one bug

Fix minor typo in Deadline and Event class

tofu-and-noodles committed:
Change while-loop condition, and shift input prompt to end of loop

Change while-loop condition from while Duke isActive to
scanner.hasNext(), as the former condition throws
NoSuchElementException.

Shift the input prompt from start of while-loop to end of loop
due to the above loop condition change, as scanner.hasNext() blocks
at the condition check, thus leading to the input prompt incorrectly
appearing after the input has been given.

tofu-and-noodles committed:
Update test cases in input.txt, and expected output in EXPECTED.TXT

The input test cases tests for all the necessary functionalities as
defined in Levels 1 to 4. Test cases do not test for unsupported
input (eg. "event assignment", where the expected "/at time" is not
provided.

tofu-and-noodles committed:
Shift while-loop to outside of listen()

listen() now takes in a string argument to carry out the command.
It makes more sense for listen() to be responsible only for command
execution, without handling any of the while-loop logic. while-loop
logic should be handled by the caller (Duke) instead.

tofu-and-noodles committed:
Add Exception for each of the Task subclasses

The above change is to facilitate customized handling of the errors
that arise from each of the Task subclasses.

tofu-and-noodles committed:
Add exception handling to Duke class

All the Task subclasses throw their corresponding exceptions, with
each of the exceptions handled by the Duke object.

tofu-and-noodles committed:
Duke class: Add delete function

Add the capability to delete previously added tasks from the list.

tofu-and-noodles committed:
DukeException subclasses: Unify all subclasses as DukeException

All DukeException subclasses seem redundant at the moment.
To account for the removal, exception handling in Duke.listen() has
been modified.

tofu-and-noodles committed:
Add fluff

tofu-and-noodles committed:
Change Task to abstract

tofu-and-noodles committed:
Add save functionality

Add .toLog() method to all tasks to support save functionality by
print each task in save string format.

tofu-and-noodles committed:
Change date in Event and Deadline classes

Change date in Event and Deadline classes from String to
LocalDateTime.

tofu-and-noodles committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

tofu-and-noodles committed:
Merge with Branch-Level-7

tofu-and-noodles committed:
Add classes and rework existing classes to support more OOP.

Add new Command class, with a subclass for each valid command type.
Add Parser class to process user input into Command.
Add Storage class to handle save function.
Add TaskList class to handle task lists.
Add Ui class to handle user interface actions such as reading input
and printing output.

tofu-and-noodles committed:
TaskList: Add findTasks() method to support finding matching tasks

tofu-and-noodles committed:
Add JUnit tests for ToDo, Deadline, Event, TaskList

tofu-and-noodles committed:
Add load function to Storage class

Add the ability to load tasks saved in an existing save file.

tofu-and-noodles committed:
Add load function to Storage class

Add the ability to load tasks saved in an existing save file.

tofu-and-noodles committed:
Add Javadoc and change codes to be compliant with CS2103 Checkstyle

tofu-and-noodles committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/Storage.java

tofu-and-noodles committed:
Edit to comply with CS2103 Checkstyle

yhtMinceraft1010X commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,8 @@
+/**
+ * Represents a valid Command for the Duke object.
+ */
+
```
Shouldn't the last line of the Javadoc comment be right before the Command class, rather than have a line space in between both?

yhtMinceraft1010X commented on src/main/java/CommandBye.java:
```diff
@@ -0,0 +1,26 @@
+ */
+public class CommandBye extends Command {
+    /**
+     * Closes the provided attributes of Duke object in preparation of ending the run.
```
Perhaps an empty line between the description and parameters would fulfil the code's adherence to standards regarding Javadoc? ... I've also noticed the same trait across many Javadoc comments in most classes

yhtMinceraft1010X commented on src/main/java/CommandDelete.java:
```diff
@@ -0,0 +1,20 @@
+public class CommandDelete extends Command {
```
Shouldn't there be Javadoc comments on the CommandDelete class and its public methods? ...I've also noticed a few other classes which are missing Javadoc comments.

yhtMinceraft1010X commented on src/main/java/CommandDelete.java:
```diff
@@ -0,0 +1,20 @@
+
+    @Override
+    public String toDukeOutput() {
+        return "Alrighty bossman. I shall wipe this task off the face of the earth: ";
```
I also noticed that many one-line public methods from this point on also lack Javadoc comments. I believe that there should still be Javadoc for one-line methods to explain what the method intends to achieve.

yhtMinceraft1010X commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,54 @@
+            }
+        }
+    }
+    //    private void shutDown() {
```
Perhaps this unused method should be removed?

yhtMinceraft1010X reviewed (commented):
Overall a very well organised code, although I see more potential benefit from adding Javadoc comments at all public methods and classes, inclusive of one-liners.

simran-bhadani3 commented on src/main/java/CommandDeadline.java:
```diff
@@ -0,0 +1,45 @@
+ * the Deadline Task with the specified attributes into the provided TaskList.
+ */
+public class CommandDeadline extends Command {
+    private final String description;
```
Final variables should be written using uppercase.

simran-bhadani3 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,103 @@
+     * Saves the tasks in taskList into save file "duke_saved_tasks" in the same directory.
+     * @param taskList TaskList containing the tasks to be saved.
+     */
+    public void save(TaskList taskList) {
```
Perhaps you can be slightly more descriptive here and explain what exactly is being saved.

simran-bhadani3 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,103 @@
+            File file = new File("duke_saved_tasks");
+            boolean isAlreadyPresent = !file.createNewFile();
+
+            if (isAlreadyPresent) {
```
Perhaps you can rename this to isFileAlreadyPresent to make it more clear what exactly the boolean checks for.

simran-bhadani3 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,103 @@
+                FileReader fileReader = new FileReader(file);
+                BufferedReader bufferedReader = new BufferedReader(fileReader);
+                String currLine = bufferedReader.readLine();
+                ArrayList<Task> ret = new ArrayList<>();
```
I feel that you can rename ret to something more descriptive which clearly indicates what the array list is supposed to hold.

simran-bhadani3 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,37 @@
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+    protected final String icon;
```
Icon field should be in uppercase since it is a constant.

simran-bhadani3 reviewed (commented):
Classes and methods are in general well named. You can be slightly more descriptive in your naming.

tofu-and-noodles committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

tofu-and-noodles committed:
Add gradle support to project

tofu-and-noodles committed:
Set up Gradle Checkstyle and edit Javadoc comments.

tofu-and-noodles committed:
Change UI from text to graphical

tofu-and-noodles committed:
Add exit on bye command

tofu-and-noodles committed:
Add handling of IndexOutOfBoundsException to done and delete

tofu-and-noodles committed:
Add assertions

tofu-and-noodles committed:
Refactor Parse.parse() to comply with SLAP.

schping committed:
Merge pull request #1 from oeiyiping/branch-A-Assertions

Add assertions

tofu-and-noodles committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/Storage.java
#	src/main/java/TaskList.java

tofu-and-noodles committed:
Add TimedTask class which extends from Task class

Refactor Event and Deadline class to extend from TimedTask.

tofu-and-noodles committed:
Add Update command to Duke

Update command changes the specified fields of the task specified by
the input index.

schping committed:
Merge pull request #2 from oeiyiping/branch-C-Update

Add Update command to Duke

tofu-and-noodles committed:
Update README to include introduction and features of the bot

tofu-and-noodles committed:
Add sorting of Task, by date created, description, isDone, icon

tofu-and-noodles committed:
Rename all Duke to Henchman

tofu-and-noodles committed:
Search term in TaskList.findTasks() is now case-insensitive
Update README

tofu-and-noodles committed:
Update main README

tofu-and-noodles committed:
Add app title and icon

tofu-and-noodles committed:
Add screenshots of app, edit README

tofu-and-noodles committed:
Edit README to correct format

tofu-and-noodles committed:
Add Ui.png

tofu-and-noodles committed:
Update README with an example picture

tofu-and-noodles committed:
Change image format in README

tofu-and-noodles committed:
Change all jpg to png

tofu-and-noodles committed:
Add image into README

tofu-and-noodles committed:
Add image into README

tofu-and-noodles committed:
Change image size

tofu-and-noodles committed:
Add all images into README

tofu-and-noodles committed:
Add linebreaks to README for better formatting

tofu-and-noodles committed:
Add more linebreaks for better formatting

tofu-and-noodles committed:
Further format README

tofu-and-noodles committed:
Change mainclass in build.gradle to build proper jar

tofu-and-noodles committed:
Add JavaDoc for public classes and methods

tofu-and-noodles committed:
Capitalize all constants

tofu-and-noodles committed:
Format code to comply with checkstyle

tofu-and-noodles committed:
Remove irrelevant test cases

schping committed:
Set theme jekyll-theme-cayman

schping committed:
Set theme jekyll-theme-slate

schping committed:
Set theme jekyll-theme-cayman

schping committed:
Update README.md
